builtInFunctions/esdtDataStorage.go

```diff
@@ -429,7 +467,11 @@ func (e *esdtDataStorage) marshalAndSaveData(
 	return e.accounts.SaveAccount(systemAcc)
 }
 
-func (e *esdtDataStorage) getSystemAccount() (vmcommon.UserAccountHandler, error) {
+func (e *esdtDataStorage) getSystemAccount(options queryOptions) (vmcommon.UserAccountHandler, error) {
+	if options.isCustomSystemAccountSet && !check.IfNil(options.customSystemAccount) {
+		return options.customSystemAccount, nil
+	}
+
 	systemSCAccount, err := e.accounts.LoadAccount(vmcommon.SystemAccountAddress)
```

Comment:
suggestion to extract L475:L486 into a function called loadSystemAccount()
So, the getSystemAccount will act as a proxy functoin